src/stamina/_core.py

```diff
@@ -586,9 +587,11 @@ def _compute_backoff(
     """
     if CONFIG.testing is not None:
         return 0.0
-
+    if not initial:
+        return 0.0
+    if max_backoff and math.log(max_backoff / initial, exp_base) < num - 1:
```

Comment:
Could you please add a comment explaining this short circuit calculation (and return our empty lines 😅)?